### ISSUE TITLE
allow vactube random spinning to be disabled or alternative spinning styles

### ISSUE THREAD

**Issue by Fennecai** [TeamSpen210/HammerAddons#159]
I've seen some maps where the cubes and stuff going through vactubes move smoothly through them without a ton of random spinning similar to portal 1; probably as a result of using the old path_track method, but this is something that looks aesthetically pleasing in some places.

it would be cool if there was an option to change the way the vactube items rotate and stuff, between choices like "randomized (default) / smooth (rotate randomly along movement axis, but face movement direction) / fully smooth (no random rotation, only face movement direction)"

or simply a boolean to disable the randomized spinning and only face the movement direction.

possibly also, there could be an option per-junction to do the same thing, so that the cubes might slide through smoothly in one section and bump around violently in another.

**Comment by TeamSpen210** [comment]
This is true, it needs to be controllable, and I also need to make it influenced by the actual movement of the path instead of being completely random.

**Comment by Fennecai** [comment]
if your going to overhaul the rotations i might as well mention another idea i had, which is to check the bounding-box of a model and if one axis of it exceeds 120 units or so, then it will not rotate the model in a way that is likely to clip through straight tubes. (cant control that with curves for obvious geometrical reasons.)

I've wanted to have vactubes with aperture desks in them at times and this would help- probably a bit too niche to consider though....
